driver/registry_default_registration.go

```diff
@@ -8,17 +8,30 @@ import (
 	"github.com/ory/kratos/selfservice/flow/registration"
 )
 
+func contains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func filter(hooks []config.SelfServiceHook, persistencePhase string) (ret []config.SelfServiceHook) {
+	for _, h := range hooks {
+		if len(h.PersistencePhase) == 0 || contains(h.PersistencePhase, persistencePhase) {
```

Comment:
you should not check for `len(h.PersistencePhase)` since it will be handled by `contains()` if you set default value for `PersistencePhase` in schema. I think this should work and be more logical.